internal/core/monitoring/metrics.go

```diff
@@ -0,0 +1,95 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/topfreegames/maestro/internal/config"
+	"go.uber.org/zap"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type MetricOpts struct {
+	Namespace  string
+	Subsystem  string
+	Name       string
+	Help       string
+	Labels     []string
+	Buckets    []float64
+	Objectives map[float64]float64
+}
+
+// runMetricsServer start a metrics server in other goroutine, and returns a
+// shutdown function.
+func RunMetricsServer(ctx context.Context, configs config.Config) func() error {
```

Comment:
Not sure about the metrics server initialization being here. It seems like this is the responsibility of the `cmd` or its supportive packages (like `internal/services`). WDYT?

Reply:
Moved to internal/services/metrics.go